### PR TITLE
'feat: quantile function added with its test cases'

### PR DESCRIPTION
**Pull request by WargnerM**

### CODE REVIEW

File: src/quantile.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-unused-vars */
```
Comment (lpatiny) [inline src/quantile.js]: You should never remove eslint checks but rather fix the error !

File: src/quantile.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * @param {object} data - x sorted in increasing order and y the number of occurrence of each x value
+ * @param {numbers} k, q - the kth q-quantile, thus k should be less than or equal to q, should be positive integers
```
Comment (lpatiny) [inline src/quantile.js]: Why not only one parameters between 0,1 like in mathlab ? https://www.mathworks.com/help/stats/quantile.html
Please also use full name for variables and not just 'p'

File: src/quantile.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  let p = k / q;
+
+  if (p === 1) {
+    return x[x.length - 1];
+  }
+
+  let sumY = 0;
+  let i;
```
Comment (lpatiny) [inline src/quantile.js]: Please define i directly in the loop
`for (let i = 0, ...`

File: src/quantile.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  for (i = 0; i < y.length; i++) {
+    sumY += y[i];
+  }
+
+  let index = (sumY - 1) * p;
+  let cumY = 0;
+
+  for (i = 0; i < y.length; i++) {
```
Comment (lpatiny) [inline src/quantile.js]: Redefine `let i` in the loop here as well

File: src/__tests__/quantile.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  it('second quartile of four consecutive number with occurrence 1:2:1:2, should return 2.5', () => {
+    const data = {
+      x: [1, 2, 3, 4],
+      y: [1, 2, 1, 2],
+    };
+    const k = 1;
+    const q = 2;
+    expect(quantile(data, k, q)).toStrictEqual(2.5);
+  });
+});
```
Comment (lpatiny) [inline src/__tests__/quantile.test.js:46]: Please add one more example with 1,2,2,1. Should also give 2.5 right ?

File: src/quantile.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  let sumY = 0;
+  let i;
+
+  for (i = 0; i < y.length; i++) {
+    sumY += y[i];
+  }
+
+  let index = (sumY - 1) * p;
```
Comment (lpatiny) [inline src/quantile.js]: You should make the Math.floor() + 1 at this level and not recalculate it continuously in the loop. Seems like a waste of time.

**Review by lpatiny — CHANGES_REQUESTED** [review]

File: src/quantile.js (hunk 2 above)
Comment (WargnerM) [reply src/quantile.js]: Good idea, I have changed to cumProb (cumulative probability)

File: src/quantile.js (hunk 3 above)
Comment (WargnerM) [reply src/quantile.js]: Done in the new code

File: src/quantile.js (hunk 4 above)
Comment (WargnerM) [reply src/quantile.js]: The variable is already redefined on the new version

File: src/__tests__/quantile.test.js (hunk 5 above)
Comment (WargnerM) [reply src/__tests__/quantile.test.js:46]: The result is correct, It is also included

File: src/quantile.js (hunk 6 above)
Comment (WargnerM) [reply src/quantile.js]: the code needs the fractional part for the interpolation, so instead I've decided to create a new variable index2

**Pull request closed without merging**